instrumentation/jmx-metrics/javaagent/src/test/java/io/opentelemetry/instrumentation/javaagent/jmx/JmxMetricInsightInstallerTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.javaagent.jmx;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.instrumentation.jmx.yaml.JmxConfig;
+import io.opentelemetry.instrumentation.jmx.yaml.JmxRule;
+import io.opentelemetry.instrumentation.jmx.yaml.RuleParser;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class JmxMetricInsightInstallerTest {
+  private static final String PATH_TO_ALL_EXISTING_RULES = "src/main/resources/jmx/rules";
+  private static RuleParser parser;
+  private static final Set<String> FILES_TO_BE_TESTED =
+      new HashSet<>(
+          Arrays.asList(
+              "activemq.yaml",
+              "hadoop.yaml",
+              "jetty.yaml",
+              "kafka-broker.yaml",
+              "tomcat.yaml",
+              "wildfly.yaml"));
+
+  @BeforeAll
+  static void setup() {
+    parser = RuleParser.get();
+    assertThat(parser == null).isFalse();
+  }
+
+  @Test
+  void testToVerifyExistingRulesAreValid() throws Exception {
+    File existingRulesDir = new File(PATH_TO_ALL_EXISTING_RULES);
```

Comment:
Rather than opening these as files, WDYT about using loading these as resources instead? (`Class#getResource()`)

Using the `File` API places an implicit dependency on the current working directory -- sure, gradle currently runs the tests with the current module's root as cwd, and that's probably not likely to change in the future, but the less preconditions we have in these tests the less surprises we'll get in the future.

Reply:
Yeah, I did try using getResource() originally. Then I had trouble to get to the `resources`. (I think it expects those files to be in `test/resources` folder.) Then I think test case probably not going to move around (and it is probably a bad idea to duplicate all those files), so I decided to use the `File` to load. Let me know if you knew how to refer to those resources files. I tried relative path from the test case, tried from the jmx-metric level etc. No luck.

Reply:
> Yeah, I did try using getResource() originally. Then I had trouble to get to the `resources`. (I think it expects those files to be in `test/resources` folder.)

Did you try the full `src/main/resources/jmx/rules` path? Or just `jmx/rules`? The `getResource()` API loads these file from the jar file, the `src/main/resources/` prefix is no longer there at that point

Reply:
> > Yeah, I did try using getResource() originally. Then I had trouble to get to the `resources`. (I think it expects those files to be in `test/resources` folder.)
> 
> Did you try the full `src/main/resources/jmx/rules` path? Or just `jmx/rules`? The `getResource()` API loads these file from the jar file, the `src/main/resources/` prefix is no longer there at that point

Yes, I tried. `src/main/resources/jmx/rules/tomcat.yaml`, `/src/main/resources/jmx/rules/tomcat.yaml`,`/jmx/rules/tomcat.yaml`,`jmx/rules/tomcat.yaml`, `../../../../../../../../resources/jmx/rules/tomcat.yaml`, none of them worked

Reply:
I tested it myself, and it's true, that does not work (surprisingly).

Just for documentation's sake, the culprit behind this behavior is here: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/0f258c669aeda3aec3fa84725bc6e75ab7b35da6/conventions/src/main/kotlin/io.opentelemetry.instrumentation.javaagent-testing.gradle.kts#L120-L124

We don't want to change that behavior -- using `File` should be perfectly fine here. Thanks for your patience with me @shuwpan !

Reply:
Thanks @mateuszrzeszutek for taking the time to review and find the culprit. According to my google search, it should've worked too. I was suspecting something in the build file causing this, but it is too daunting to dig in so I didn't go all the way to figure out why. Thanks for pointing it out.